app/actions/checkPhase.ts

```diff
@@ -0,0 +1,25 @@
+import * as moment from "moment";
+import { IcoParametersState, selectEndDate, selectStartDate } from "../reducers/icoParameters";
+import { IcoPhase, NEW_PHASE_ACTION } from "./constants";
+
+export function changePhaseAction(newPhase: IcoPhase) {
+  return {
+    type: NEW_PHASE_ACTION,
+    payload: newPhase,
+  };
+}
+export function checkPhase(icoParams: IcoParametersState) {
```

Comment:
I really dont like that we rely on this function in reducer code. They are not pure anymore then. What about making it a selector?

Reply:
and not putting this value into the store at all?

Reply:
Lets discuss it.